crates/core/src/rpc/surfnet_cheatcodes.rs

```diff
@@ -743,4 +754,82 @@ impl SvmTricksRpc for SurfnetCheatcodesRpc {
             })
         })
     }
+
+    fn get_transaction_profile(
+        &self,
+        meta: Self::Metadata,
+        transaction_signature: UuidOrSignature,
+    ) -> BoxFuture<Result<RpcResponse<ProfileResult>>> {
+        let svm_locker = match meta.get_svm_locker() {
+            Ok(locker) => locker,
+            Err(e) => return e.into(),
+        };
+        let first_slot = svm_locker
+            .get_first_local_slot()
+            .unwrap_or_else(|| svm_locker.with_svm_reader(|svm| svm.get_latest_absolute_slot()));
+        let latest_slot = svm_locker.with_svm_reader(|svm| svm.get_latest_absolute_slot());
+
+        Box::pin(async move {
+            let mut profile_result =
+                svm_locker.with_svm_reader(|svm| match &transaction_signature {
+                    UuidOrSignature::Signature(signature) => {
+                        svm.executed_transaction_profiles.get(signature).cloned()
+                    }
+                    UuidOrSignature::Uuid(uuid) => {
+                        svm.simulated_transaction_profiles.get(uuid).cloned()
+                    }
+                });
+
+            if profile_result.is_none() {
+                match &transaction_signature {
+                    UuidOrSignature::Signature(signature) => {
+                        let tx_opt = svm_locker.with_svm_reader(|svm| {
+                            svm.transactions
+                                .get(signature)
+                                .map(|entry| entry.expect_processed().transaction.clone())
+                        });
+                        if let Some(transaction) = tx_opt {
+                            if let Ok(SvmAccessContext { inner: prof, .. }) = svm_locker
+                                .profile_transaction(
+                                    &None,
+                                    transaction,
+                                    Some(UiAccountEncoding::JsonParsed),
+                                )
+                                .await
+                            {
+                                profile_result = Some(prof);
+                            }
+                        }
+                    }
+                    UuidOrSignature::Uuid(_) => {
+                        // No reconstruction for UUID, error will be returned below if not found
+                    }
+                }
+            }
```

Comment:
This is interesting, but I'm not sure if we should have it. If I'm getting a profile result for a signature and we don't have results, I'd just expect it to return `None`, not simulate.

Reply:
Thoughts @lgalabru?